docs/architecture/blazor-for-web-forms-developers/pages-routing-layouts.md

```diff
@@ -0,0 +1,265 @@
+
+@code {
+    string text;
+    string name;
+
+    void OnClick() {
+        name = text;
+    }
+}
+```
+
+## Create pages
+
+To create a page in Blazor, create a component and add the `@page` Razor directive to specify the route for the component. The `@page` directive takes a single parameter, which is the route template to add to that component.
+
+```razor
+@page "/counter"
+```
+
+The route template parameter is required. Unlike ASP.NET Web Forms, the route to a Blazor component *isn't* inferred from its file location (although that may be a feature added in the future).
+
+The route template syntax is the same basic syntax used for routing in ASP.NET Web Forms. Route parameters are specified in the template using braces. Blazor will bind route values to component parameters with the same name (case-insensitive).
+
+```razor
+@page "/product/{id}"
+
+<h1>Product @Id</h1>
+
+@code {
+    [Parameter]
+    public string Id { get; set; }
+}
+```
+
+You can also specify constraints on the value of the route parameter. For example, to constrain the product ID to be an `int`:
+
+```razor
+@page "/product/{id:int}"
+
+<h1>Product @Id</h1>
+
+@code {
+    [Parameter]
+    public int Id { get; set; }
+}
+```
+
+For a full list of the route constraints supported by Blazor, see [Route constraints](/aspnet/core/blazor/routing#route-constraints).
+
+## Router component
+
+Routing in Blazor is handled by the `Router` component. The `Router` component is typically used in the app's root component (*App.razor*).
+
+```razor
+<Router AppAssembly="@typeof(Program).Assembly">
+    <Found Context="routeData">
+        <RouteView RouteData="@routeData" DefaultLayout="@typeof(MainLayout)" />
+    </Found>
+    <NotFound>
+        <LayoutView Layout="@typeof(MainLayout)">
+            <p>Sorry, there's nothing at this address.</p>
+        </LayoutView>
+    </NotFound>
+</Router>
+```
+
+The `Router` component discovers the routable components in the specified `AppAssembly` and in the optionally specified `AdditionalAssemblies`. When the browser navigates, the `Router` intercepts the navigation and renders the contents of its `Found` parameter with the extracted `RouteData` if a route matches the address, otherwise the `Router` renders its `NotFound` parameter.
+
+The `RouteView` component handles rendering the matched component specified by the `RouteData` with its layout if it has one. If the matched component doesn't have a layout, then the optionally specified `DefaultLayout` is used.
+
+The `LayoutView` component renders its child content within the specified layout. We'll look at layouts more in detail later in this chapter.
+
+## Navigation
+
+In ASP.NET Web Forms, you trigger a navigation to a different page by returning a redirect response to the browser. Returning a redirect response isn't typically possible in Blazor. Blazor doesn't use a request-reply model. You can, however, trigger browser navigations directly, just like you can with JavaScript.
```

Comment:
I suggest providing a basic `Response.Redirect(...);` example here for Web Forms. Maybe something like the following from a code-behind:

```csharp
        protected void NavigateButton_Click(object sender, EventArgs e)
        {
            Response.Redirect("Counter");
        }
```

Reply:
@danroth27 I'll take care of this. I'm reviewing this PR right now.